model/dao/ProjectDAO/PostgreSQLProjectDAO.php

```diff
@@ -669,27 +669,42 @@ public function create(ProjectVO $projectVO) {
      * This function deletes the data of a Project by its {@link ProjectVO}.
      *
      * @param ProjectVO $projectVO the {@link ProjectVO} with the data we want to delete from database.
-     * @return int the number of rows that have been affected (it should be 1).
-     * @throws {@link SQLQueryErrorException}
+     * @return OperationResult the result {@link OperationResult} with information about operation status
      */
     public function delete(ProjectVO $projectVO) {
-        $affectedRows = 0;
+        $result = new OperationResult(false);
 
-        // Check for a user ID.
-        if($projectVO->getId() >= 0) {
-            $currProjectVO = $this->getById($projectVO->getId());
-        }
+        $sql = "DELETE FROM project WHERE id=:id";
+        try {
+            $statement = $this->pdo->prepare($sql);
+            $statement->bindValue(":id", $projectVO->getId(), PDO::PARAM_INT);
+            $statement->execute();
 
-        // If it exists, then delete.
```

Comment:
We used to check if the project existed before removing it, although we did it badly and caused a crash. We never expected to catch this error in upper layers.

In PDO migrations I've been removing such a check. The SQL query will just run and do nothing, with no errors.